lib/msal-browser/src/cache/TokenCache.ts

```diff
@@ -109,19 +129,21 @@ export class TokenCache implements ITokenCache {
         } else {
             throw BrowserAuthError.createUnableToLoadTokenError("loadExternalTokens is designed to work in browser environments only.");
         }
+
+        return new CacheRecord(accountEntity, idTokenEntity);
     }
 
     /**
      * Helper function to load access tokens to msal-browser cache
      * @param request
      * @param response
      * @param options
-     * @param homeAccountId
+     * @param cacheRecord
      * @param environment
      * @param tenantId
      * @returns
      */
-    private loadAccessToken(request: SilentRequest, response: ExternalTokenResponse, homeAccountId: string, environment: string, tenantId: string, options: LoadTokenOptions): void {
+    private loadAccessToken(request: SilentRequest, response: ExternalTokenResponse, cacheRecord: CacheRecord, environment: string, tenantId: string, options: LoadTokenOptions): void {
```

Comment:
I assume the `SilentRequest` object should contain the account info needed to create the `AccessTokenEntity` later in this method. If not, it'd be better to generate the homeAccountId before generating the entities so one doesn't depend on the other.

Reply:
Right now it generates both the AccountEntity and the IdTokenEntity in the loadIdToken() method. There were already a few different ways for it to be generated, but I didn't fundamentally change any of that.

Reply:
I created a new `loadAccount` helper function that makes it a bit clearer.